python/lsst/summit/utils/guiders/transformation.py

```diff
@@ -0,0 +1,714 @@
+    return rotation
+
+
+def mk_ccd_to_dvcs(
+    bbox_ccd: Box2I,
+    det_nquarter: int,
+) -> tuple[AffineTransform, AffineTransform]:
+    """
+    Make transformations suitable for Guider stamps, to go from a view of the
+    stamp in CCD pixel coordinates (ie. with 0,0 in the Lower Left of C00) to
+    DVCS view (ie. orientated in the Focal Plane correctly in the DVCS, with
+    0,0 at the Lower Left of the ROI)
+
+    Parameters
+    ----------
+    bbox_ccd : Box2I
+        Bounding Box for the ROI in CCD pixel coordinates
+    det_nquarter : int
+        The number of 90degree quarters needed to rotate from CCD view to DVCS
+        view
+
+    Returns
+    -------
+    forwards : AffineTransform
+        A transform from CCD pixel coordinates in the CCD view to Stamp pixel
+        coordinates in DVCS view
+
+    backwards : AffineTransform
+        A transform from Stamp pixel coordinates in DVCS view to CCD pixel
+        coordinates in the CCD view
+
+    Example
+    -------
+
+    detector = camera[189] bbox_ccd = Box2I(Point2I(300,600),Point2I(350,650))
+    ft,bt = mk_ccd_to_dvcs(box_ccd,detector.getOrientation().getNQuarter())
+
+    # given a sky coordinate, find point on the stamp in DVCS view pt_ccd =
+    wcs.skyToPixel(sky_coord) pt_stamp_dvcs = ft(pt_ccd)
+
+    # given a point on the stamp in DVCS view, find the skyCoord pt_ccd =
+    bt(pt_stamp_dvcs) sky_coord = wcs.pixelToSky(pt_ccd)
+
+    """
+    # build integer rotation matrices
+    rot = {}
+    rot[0] = np.array([[1.0, 0.0], [0.0, 1.0]])
+    rot[1] = np.array([[0.0, -1], [1.0, 0.0]])
+    rot[2] = np.array([[-1.0, 0.0], [0.0, -1.0]])
+    rot[3] = np.array([[0.0, 1.0], [-1.0, 0.0]])
+
+    irot = {}
+    irot[0] = rot[0].transpose()
+    irot[1] = rot[1].transpose()
+    irot[2] = rot[2].transpose()
+    irot[3] = rot[3].transpose()
+
+    # number of 90deg CCW rotations
+    nq = np.mod(det_nquarter, 4)
+
+    # get LL,Size of the CCD view BBox
+    llpt_ccd = Extent2D(bbox_ccd.getCorners()[0])
+    nx, ny = bbox_ccd.getDimensions()
+
+    # get translations to use for each NQ value
+    boxtranslation = {}
+    boxtranslation[0] = Extent2D(0, 0)
+    boxtranslation[1] = Extent2D(ny - 1, 0)
+    boxtranslation[2] = Extent2D(nx - 1, ny - 1)
+    boxtranslation[3] = Extent2D(0, nx - 1)
+
+    # build the forware Transform
+    ftranslation = AffineTransform.makeTranslation(-llpt_ccd)
+    frotation = AffineTransform(rot[nq])
+    fboxtranslation = AffineTransform.makeTranslation(boxtranslation[nq])
+    # ordering is third*second*first
+    forwards = fboxtranslation * frotation * ftranslation
+
+    btranslation = AffineTransform.makeTranslation(llpt_ccd)
+    brotation = AffineTransform(irot[nq])
+    bboxtranslation = AffineTransform.makeTranslation(-boxtranslation[nq])
+    backwards = btranslation * brotation * bboxtranslation
+
+    return forwards, backwards
+
+
+def mk_roi_bboxes(md, camera):
```

Comment:
Rename to `make_roi_bboxes` or something like that. Also it only returns a single box so don't use boxes plural.

Reply:
missing mypy annotations, including return type